homeassistant/components/ezviz/camera.py

```diff
@@ -1,197 +1,253 @@
-        self._privacy = data["privacy"]
-        self._audio = data["audio"]
-        self._ir_led = data["ir_led"]
-        self._state_led = data["state_led"]
-        self._follow_move = data["follow_move"]
-        self._alarm_notify = data["alarm_notify"]
-        self._alarm_sound_mod = data["alarm_sound_mod"]
-        self._encrypted = data["encrypted"]
-        self._local_ip = data["local_ip"]
-        self._detection_sensibility = data["detection_sensibility"]
-        self._device_sub_category = data["device_sub_category"]
-        self._local_rtsp_port = data["local_rtsp_port"]
-
-    async def async_added_to_hass(self):
-        """Subscribe to ffmpeg and add camera to list."""
-        self._ffmpeg = self.hass.data[DATA_FFMPEG]
-
-    @property
-    def should_poll(self) -> bool:
-        """Return True if entity has to be polled for state.
+        camera_entities.append(
+            EzvizCamera(
+                hass,
+                coordinator,
+                idx,
+                camera_username,
+                camera_password,
+                camera_rtsp_stream,
+                local_rtsp_port,
+                ffmpeg_arguments,
+            )
+        )
 
-        False if entity pushes its state to HA.
-        """
-        return True
+    async_add_entities(camera_entities)
+
+
+class EzvizCamera(CoordinatorEntity, Camera, RestoreEntity):
+    """An implementation of a Ezviz security camera."""
+
+    def __init__(
+        self,
+        hass,
+        coordinator,
+        idx,
+        camera_username,
+        camera_password,
+        camera_rtsp_stream,
+        local_rtsp_port,
+        ffmpeg_arguments,
+    ):
+        """Initialize a Ezviz security camera."""
+        super().__init__(coordinator)
+        Camera.__init__(self)
+        self._username = camera_username
+        self._password = camera_password
+        self._rtsp_stream = camera_rtsp_stream
+        self._idx = idx
+        self._ffmpeg = hass.data[DATA_FFMPEG]
+        self._local_rtsp_port = local_rtsp_port
+        self._ffmpeg_arguments = ffmpeg_arguments
+
+        self._serial = self.coordinator.data[self._idx]["serial"]
+        self._name = self.coordinator.data[self._idx]["name"]
+        self._local_ip = self.coordinator.data[self._idx]["local_ip"]
 
     @property
     def device_state_attributes(self):
         """Return the Ezviz-specific camera state attributes."""
         return {
-            # if privacy == true, the device closed the lid or did a 180° tilt
-            "privacy": self._privacy,
+            # Camera firmware version
+            "sw_version": self.coordinator.data[self._idx]["version"],
+            # Camera firmware version update available?
+            "upgrade_available": self.coordinator.data[self._idx]["upgrade_available"],
+            # if privacy == true, the device closed the lid or did a 180� tilt
+            "privacy": self.coordinator.data[self._idx]["privacy"],
```

Comment:
Binary measurements should be binary sensors.

Since this PR will be a breaking change we have the opportunity to correct the integration and have it follow our design standards.

Reply:
Done. Added platforms (sensor, binary_sensor and switch)